lib/macmillan/utils/middleware/cookie_message.rb

```diff
@@ -15,17 +15,15 @@ def initialize(app, options = {})
           @log_level = options[:log_level]
 
           if (logger = options[:logger])
-            if logger.respond_to?(:tagged)
-              @logger = logger
-            else
-              @logger = ActiveSupport::TaggedLogging.new(logger)
-            end
+            build_tagged_logger(logger)
           end
         end
 
         def call(env)
           @request = Rack::Request.new(env)
 
+          build_tagged_logger(logger)
```

Comment:
Problem is - for some apps calling this middleware (e.g. sherlock as described below) we don't have a logger in context at the point of `initialize`